dev/system-test/firestore.ts

```diff
@@ -1313,6 +1313,44 @@ describe('DocumentReference class', () => {
   });
 });
 
+describe('runs query on a large collection', () => {
+  let firestore: Firestore;
+  let randomCol: CollectionReference;
+
+  beforeEach(async () => {
+    firestore = new Firestore({});
+    randomCol = getTestRoot(firestore);
+
+    const promises: Array<Promise<DocumentReference<DocumentData>>> = [];
+    for (let i = 0; i < 1000; i++) {
+      promises.push(randomCol.add({foo: 'a'}));
+    }
+    await Promise.all(promises);
+  });
+
+  afterEach(() => verifyInstance(firestore));
+
+  it('can get()', () => {
+    return randomCol.get().then(res => {
+      expect(res.size).to.equal(1000);
+    });
+  });
+
+  it('can stream()', done => {
+    let received = 0;
+    return randomCol
```

Comment:
You shouldn't need to return the result here. Returning anything could be misleading, since it could be confused with returning a `promise`, and I think this code is returning a `Stream`.

You use the `done` callback to signal that the test is complete instead of returning a `promise`. That's fine since that pattern is used in other tests in this file.